support/scripts/mkgrubimg

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env bash
+
+# allow color coded outputs
+NC="\033[0m"
+RED="\033[0;31m"
+
+# help menu
+usage()
+{
+	echo "Usage: $0 [options] [output file]"
+	echo "Creates bootable GRUB images from Unikraft kernel images."
+	echo ""
+	echo "  -h                         Display help and exit"
+	echo "  -f                         Output format (supported: iso)"
+	echo "  -i                         (optional) Path to initrd directory"
+	echo "  -k                         Path to the Unikraft kernel image"
+	exit 1
+}
+
+# process options
+while getopts "hk:i:f:" options; do
+	case "${options}" in
+		h)
+			usage
+			;;
+		f)
+			OPTFORMAT=${OPTARG}
+			;;
+		i)
+			OPTINITRD=${OPTARG}
+			;;
+		k)
+			OPTKERNELIMG=${OPTARG}
+			BUILDDIR=${OPTKERNELIMG%/*}
+			;;
+		*)
+			usage
+			;;
+	esac
+done
+shift $((OPTIND-1))
+
+# validate arguments
+if [[ -z ${OPTFORMAT} || -z ${OPTKERNELIMG} || -z $1 ]]; then
+	usage
+fi
+
+case "${OPTFORMAT}" in
+	# generate grub iso image
+	iso)
+		# generate configuration file
+		GRUB_CFG_TXT="set default=0 \n"
+		GRUB_CFG_TXT+="set timeout=0 \n"
+		GRUB_CFG_TXT+="menuentry \"${OPTKERNELIMG##*/}\" { \n\t"
+		GRUB_CFG_TXT+="multiboot /boot/${OPTKERNELIMG##*/} /${OPTKERNELIMG##*/} console=ttyS0 \n\t"
+		if [ ! -z ${OPTINITRD} ]; then
+			GRUB_CFG_TXT+="module /boot/initrd_archive.cpio \n\t"
+		fi
+		GRUB_CFG_TXT+="boot \n"
+		GRUB_CFG_TXT+="}"
+
+		# configure grub
+		mkdir -p "${BUILDDIR}/isofiles/boot/grub"
+		echo -e ${GRUB_CFG_TXT} > "${BUILDDIR}/isofiles/boot/grub/grub.cfg"
+		if [ ! -z ${OPTINITRD} ]; then
+			MKCPIO="${BASH_SOURCE[0]%/*}/mkcpio"
+			${MKCPIO} -q "${BUILDDIR}/isofiles/boot/initrd_archive.cpio" "${OPTINITRD}"
+		fi
+		cp "${OPTKERNELIMG}" "${BUILDDIR}/isofiles/boot/"
+
+		# build grub image
+		GRUB_MODULES="part_acorn part_amiga part_apple part_bsd part_dfly part_dvh part_gpt part_plan part_sun part_sunpc normal multiboot"
```

Comment:
Do we really need all these partition table modules? We will not boot from a GPT or Apple formatted harddisk, we boot from an ISO. I think the only required is in fact `multiboot`. Maybe `normal` is also needed but I don't know what is in there. Somewhere I saw that also `iso9660` and `biosdisk` might be needed, please double check.

Reply:
You were right, only multiboot was needed. I didn't realize ISOs didn't have partitions and I guess I must have mistaken an error fixed by something else as an error fixed by adding the partition modules.